tests/plugins/test_add_help.py

```diff
@@ -0,0 +1,181 @@
+        ADD %s /%s
+        CMD blabla""" % (AddHelpPlugin.man_filename, AddHelpPlugin.man_filename))
+
+
+def test_add_help_no_help_file(request, tmpdir, docker_tasker):
+    df_content = "FROM fedora"
+    df = df_parser(str(tmpdir))
+    df.content = df_content
+
+    workflow = DockerBuildWorkflow(MOCK_SOURCE, 'test-image')
+    workflow.builder = X
+    workflow.builder.df_path = df.dockerfile_path
+    workflow.builder.df_dir = str(tmpdir)
+
+    runner = PreBuildPluginsRunner(
+        docker_tasker,
+        workflow,
+        [{
+            'name': AddHelpPlugin.key,
+            'args': {'help_file': 'help.md'}
+        }]
+    )
+    # Runner should not crash if no help.md found
+    runner.run()
+
+
+@pytest.mark.parametrize('go_md2man_check', ['binary_missing', 'fail_at_help', 'pass'])
+@pytest.mark.parametrize('go_md2man_result', ['error', 'pass'])
+def test_add_help_md2man_error(request, tmpdir, docker_tasker, go_md2man_check, go_md2man_result):
+    df_content = "FROM fedora"
+    df = df_parser(str(tmpdir))
+    df.content = df_content
+
+    workflow = DockerBuildWorkflow(MOCK_SOURCE, 'test-image')
+    workflow.builder = X
+    workflow.builder.df_path = df.dockerfile_path
+    workflow.builder.df_dir = str(tmpdir)
+
+    help_markdown_path = os.path.join(workflow.builder.df_dir, "help.md")
+    generate_help_file_md(help_markdown_path)
+
+    if go_md2man_check == 'binary_missing':
+        (flexmock(subprocess)
+             .should_receive("check_output")
+             .with_args(["go-md2man", "-h"], stderr=subprocess.STDOUT)
+             .and_raise(OSError))
+    elif go_md2man_check == 'pass':
+        (flexmock(subprocess)
+             .should_receive("check_output")
+             .with_args(["go-md2man", "-h"], stderr=subprocess.STDOUT)
+             .and_raise(subprocess.CalledProcessError(returncode=2, cmd='foo')))
+    elif go_md2man_check == 'fail_at_help':
+        (flexmock(subprocess)
+             .should_receive("check_output")
+             .with_args(["go-md2man", "-h"], stderr=subprocess.STDOUT)
+             .and_raise(subprocess.CalledProcessError(returncode=99, cmd='foo')))
+
+    if go_md2man_result == 'fail':
+        cmd = "go-md2man -in=%s -out=%s" % (help_markdown_path, AddHelpPlugin.man_filename)
+        (flexmock(subprocess)
+             .should_receive("check_output")
+             .with_args(cmd, stderr=subprocess.STDOUT, shell=True, cwd=workflow.builder.df_dir)
+             .and_raise(subprocess.CalledProcessError(returncode=99, cmd='foo')))
+
+    fake_logger = FakeLogger()
+    existing_logger = atomic_reactor.plugin.logger
+
+    def restore_logger():
+        atomic_reactor.plugin.logger = existing_logger
+
+    request.addfinalizer(restore_logger)
+    atomic_reactor.plugin.logger = fake_logger
+
+    runner = PreBuildPluginsRunner(
+        docker_tasker,
+        workflow,
+        [{
+            'name': AddHelpPlugin.key,
+            'args': {'help_file': 'help.md'}
```

Comment:
Can we parametrize this to also test that it works the same without having to specify a `help_file` location?